### PR TITLE
test: add coverage for high-ROI reports, controllers and doctypes

### PR DESCRIPTION
**Pull request by mihir-kandoi**
## What

Adds ~250 test methods across 13 files, targeting the **highest-ROI untested logic** in `develop` (identified from a Codecov gap analysis — develop currently sits at **80.64%**). Tests-only change; no source files touched.

## Why

Most of the remaining coverage gap is either trivial/defensive code or expensive integration flows that aren't worth testing. This PR focuses on the subset where a worthwhile test protects real business logic and edge cases — query-report computation, manufacturing scheduling/capacity, accounting engines, and project rollups.

## Cross-database compatibility

Every test is written to pass on **both MariaDB and PostgreSQL**:
- No raw SQL in tests — behaviour is driven through each report's `execute()`, the ORM, or the doc's own methods.
- Result rows are **sorted in Python before assertion** (PostgreSQL has no implicit row ordering).
- Currency/float comparisons use `assertAlmostEqual` / `flt`, never raw `==`.
- No `frappe.db.commit()` — relies on `ERPNextTestSuite`'s per-test rollback.

## Coverage added

| Area | File | Focus |
|---|---|---|
| **Tier 1** | `support/report/issue_summary` | grouping (Customer/Issue Type/Priority/Assigned-To), SLA status counts, chart, empty-window edge |
| | `buying/.../supplier_scorecard_variable` | ~17 metric functions (shipments, on-time/late, received/accepted, days-late, invoiced) |
| | `accounts/report/financial_ratios` | `calculate_ratio` zero-denominator guard, root-type balance accumulation, integration smoke |
| | `selling/report/sales_analytics` | Quarterly/Yearly/Weekly ranges, Order Type/Item/Territory trees, chart curve modes |
| | `accounts/.../financial_report_engine` | pure logic: formula calc, filter-expression parser, growth transformer, segment/format engines |
| **Tier 2** | `manufacturing/.../job_card` | calendar-filter normalisation, overlap/capacity algorithm, status transitions, process-loss validation |
| | `manufacturing/.../master_production_schedule` | cumulative lead-time (recursive BOM walk), demand aggregation, order-release-date math |
| | `accounts/report/budget_variance_report` | period distribution, periodicity invariants, actual-vs-budget variance, cost-center children |
| | `manufacturing/report/production_planning_report` | WO/SO/MR `based_on` branches, order-by, raw-material gathering |
| | `assets/report/fixed_asset_register` | `get_conditions` filter building, group-by Category/Location, status/date edges |
| | `projects/.../project` | `update_percent_complete` (all 4 methods + edge branches), gross-margin guard, template copy |
| | `manufacturing/.../workstation` | working-hours totals, overlap/operating-hours exceptions, holiday recursion, status colors |
| | `accounts/.../payment_entry` | tax engine: On Paid Amount / Actual / Add-Deduct, inclusive vs exclusive, `set_liability_account` |

## Status / validation

⚠️ **Draft** — these tests have **not been run locally yet**; CI (both the MariaDB and PostgreSQL server-test matrices) is the first validation pass. I'll iterate on any failures before marking ready for review. Expensive integration paths (background jobs, repost chains, depreciation GL, deep RFQ→SQ chains, bin allotment) were intentionally left out of scope and are noted in the test docstrings.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by mihir-kandoi** [comment]
Thanks @greptile-apps — addressed in `60553bba2c`. `test_show_cumulative_is_non_decreasing` now reads the report's actual per-period budget columns (in chronological order) and asserts they're monotonically non-decreasing, start at one month's budget, and end at the annual amount — so a broken `show_cumulative` would now fail. The standalone arithmetic identity is gone. (Note: this thread is a PR conversation comment, not an inline review thread, so there's nothing to mark resolved.)

File: erpnext/buying/doctype/supplier_scorecard_variable/test_supplier_scorecard_variable.py [hunk 1]
```diff
@@ -57,3 +79,209 @@ def test_path_exists(self):
...
+		supplier = create_supplier(supplier_name=frappe.generate_hash(length=10)).name
+		# qty 4, scheduled 2023-07-15, never received -> received_qty 0 < 4.
+		self._make_po(supplier, qty=4, rate=100, schedule_date="2023-07-15")
+		sc = self._scorecard(supplier)
+
+		# DATEDIFF(end_date 2023-07-31, schedule 2023-07-15) * qty 4 = 16 * 4 = 64.
+		self.assertAlmostEqual(flt(get_item_workdays(sc)), 64.0, places=2)
+		# No receipts -> delivered-late branch 0; missed branch uses
+		# (qty - received_qty) = 4 -> 16 * 4 = 64.
+		self.assertAlmostEqual(flt(get_total_days_late(sc)), 64.0, places=2)
```
[the remark this answers is not in the document]
Comment (mihir-kandoi) [reply erpnext/buying/doctype/supplier_scorecard_variable/test_supplier_scorecard_variable.py:262]: Fixed in 06b22e5db5: replaced the MariaDB-only `CustomFunction("DATEDIFF")` with the database-aware `DateDiff` added in frappe#39974, which renders `DATEDIFF` on MariaDB and date subtraction on PostgreSQL — so `get_total_days_late()`/`get_item_workdays()` work on both backends. (Per the PR note, this import resolves once frappe#39974 merges into develop.)

File: erpnext/accounts/report/budget_variance_report/test_budget_variance_report.py [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+		self.assertEqual(len(budget_period_fields), 12)
+		for field in budget_period_fields:
+			self.assertAlmostEqual(flt(row.get(field)), monthly, places=2)
+
+		# Period budget fields should sum to the annual amount as well.
+		summed = sum(flt(row.get(f)) for f in budget_period_fields)
+		self.assertAlmostEqual(summed, annual_amount, places=2)
+
+		# Chart data mirrors the report when there is data.
+		self.assertIsNotNone(chart)
```
[the remark this answers is not in the document]
Comment (mihir-kandoi) [reply erpnext/accounts/report/budget_variance_report/test_budget_variance_report.py:102]: Fixed in 2506caeecc: the period counts are now derived from the active fiscal year (`fy_months`, `fy_quarters` computed from its start/end dates) instead of hard-coded 12/4, and the even split uses `annual / fy_months`. So it adapts to any fiscal-year start or length, and the sum-to-annual invariant is unchanged. (A fiscal year is always 12 months, so a non-Jan start alone kept it at 12 — but deriving it also covers non-12-month fiscal years and documents the intent.)
